core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala

```diff
@@ -63,8 +63,11 @@ class LocalSparkCluster(
 
     /* Start the Workers */
     for (workerNum <- 1 to numWorkers) {
+      val workDir = if (Utils.isTesting) {
```

Comment:
the fix seems fine from a cursory look. Will take a closer look soon but quick question tho - why do we need to do this only for tests? Cc @tgravescs @mridulm and @Ngone51 fyi

Reply:
This change is mainly to fix the potential UTs failures. Although it seems to be applicable in non-test scenarios, but it may affect users' usage habits. For example, some log file  that we don't care in UTs will change to use a temp dir